contracts/contracts/SimpleReadAccessController/library.cairo

```diff
@@ -0,0 +1,50 @@
+%lang starknet
+
+from starkware.cairo.common.alloc import alloc
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+from starkware.starknet.common.syscalls import get_tx_info
+from starkware.cairo.common.bool import TRUE, FALSE
+
+from SimpleWriteAccessController.library import simple_write_access_controller
+
+namespace simple_read_access_controller:
+    func constructor{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        owner_address : felt
+    ):
+        simple_write_access_controller.constructor(owner_address)
+        return ()
+    end
+
+    func has_access{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        user : felt, data_len : felt, data : felt*
+    ) -> (bool : felt):
+        let (has_access) = simple_write_access_controller.has_access(user, data_len, data)
+        if has_access == TRUE:
+            return (TRUE)
+        end
+
+        let (tx_info) = get_tx_info()
```

Comment:
@archseer @taco-paco I'm not sure this will work as intended.

1. On Ethereum we check if `userAddr == tx.origin` which gives access only to EOA (we use it to check sender, so it checks if sender equals origin), meaning off-chain access. An Ethereum account (signer) can call this method directly and read, while indirectly via other contract calls it will fail (unless explicit whitelist).

2. Now on StarkNet, we check if `userAddr == tx.account` which gives access only to StarkNet accounts (we use it to check sender, so it checks if sender equals account), which are contracts on StarkNet. 

So now someone can deploy a custom account contract, and sign a transaction that will first read from the feed (successfully) and can then use that data to share it with another contract using this custom implemented account.

We should reconsider this workflow. Any thoughts?

Reply:
Thinking on it some more, it should be ok to leave as is for now:

- We can easily update the AC if necessary
- We'll ask for suggestions during audit
- Even if custom account can be used to read prices, they would be of little use as any other contract couldn't verify the data but would need to trust the user. It could accept calls from only specific custom classes of contracts, but then the downstream protocol takes a lot of risk by depending of infrastructure Chainlink can update.

Let's add a comment here:

```suggestion
        // NOTICE: access is granted to account contracts, to enable off-chain reads. The account abstraction architecture can be used to deploy a custom contract that could access the data and pass it forward to a contract, but a workflow like that would be of little use because the contract would need to trust user inputs as it can't verify the data itself. While the financial contract could accept calls from verified account contracts only (to verify data is correct), it would be a risky strategy to depend on this part of infrastructure that can easily be upgraded.
        let (tx_info) = get_tx_info()
```

Reply:
Added comment